docs/hardening.md

```diff
@@ -0,0 +1,112 @@
+# Cluster Hardening
+
+If you want to improve the security on your cluster and make it compliant with the [CIS Benchmarks](https://learn.cisecurity.org/benchmarks), here you can find a configuration to harden your **Kubernetes** installation.
+
+To apply the hardening configuration, create a file (eg. `hardening.yaml`) and paste the content of the following code snippet into that.
+
+**N.B.** Some of these configurations have just been added to **Kubespray**, so ensure that you have the latest version to make it works properly.
+
+Also, ensure that other configurations doesn't override these.
+
+`hardening.yaml`:
+
+```yaml
+# Hardening
+---
+# This allow us to use PodSecurity admission plugin which is provided as built-in
+kube_version: v1.23.6
+
+## kube-apiserver
+authorization_modes: ['Node','RBAC']
+kube_apiserver_feature_gates: ['AppArmor=true']
+kube_apiserver_request_timeout: 120s
```

Comment:
The default value is `1m0s`
Is there any reason why we need to increase this value for hardening?

Reply:
We decided to set this value to `120s` for accessibility reasons.
As mentioned here: https://docs.datadoghq.com/security_platform/default_rules/cis-kubernetes-1.5.1-1.2.26/
> By default, it is set to 60 seconds which might be problematic on slower connections making cluster resources inaccessible once the data volume for requests exceeds what can be transmitted in 60 seconds. But, setting this timeout limit to be too large can exhaust the API server resources making it prone to Denial-of-Service attack.

We tought that `120s` could have been a good compromise to avoid cluster inaccessibility with slower connections and avoid DoS attacks. This is based on our working experience where Kubernetes is not in the internal infrastructure but is served as a remote service for the customers.

Reply:
Thanks for your explanation.
I see the reason, please keep here.